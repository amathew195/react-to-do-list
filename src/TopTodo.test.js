import React from "react";
import { fireEvent, render } from "@testing-library/react";
import TopTodo from "./TopTodo";

const top1 = {
  id: 1,
  title: 'top1',
  priority: 1,
  description: 'mock1',
}

const top2 = {
  id: 2,
  title: 'top2',
  priority: 2,
  description: 'mock2',
}

const top3 = {
  id: 3,
  title: 'top3',
  priority: 3,
  description: 'mock3',
}

const top4 = {
  id: 4,
  title: 'top4',
  priority: 1,
  description: 'mock4',
}

const todos = [top1, top2, top3];

it("renders", function() {
  render(<TopTodo todos={todos}/>)
});

describe("snapshots", function() {
  it("matches snapshot", function() {
      const { asFragment } = render(<TopTodo todos={todos}/>);
      expect(asFragment()).toMatchSnapshot();
  });
});

it("chooses correct top todo", function() {
  const {container} = render(<TopTodo todos={todos}/>)
  expect(container).toContainHTML('top1', '(priority: 1)', 'mock1');
});