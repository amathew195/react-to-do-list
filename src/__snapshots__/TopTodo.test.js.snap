// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshots matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="Todo"
  >
    <div>
      <b>
        top1
      </b>
       
      <small>
        (priority: 1)
      </small>
    </div>
    <div>
      <small>
        mock1
      </small>
    </div>
  </div>
</DocumentFragment>
`;
